mjpc/tasks/humanoid/skateboard/steering.h

```diff
@@ -51,6 +51,31 @@ class Steering : public Task {
     friend class Steering;
     int current_mode_;
     double reference_time_;
+
+    //  ============  states updated in Transition()  ============
+    // std::array<mjtNum, 3> skateboard_position_ = {0, 0, 0};
+    // std::array<mjtNum, 2> skateboard_heading_ = {0, 0};
+    // std::array<mjtNum, 2> goal_position_ = {0, 0};
+
+    //  ============  constants, computed in Reset()  ============
+    int goal_body_id_ = -1;
```

Comment:
The downside of having things under residual is that the attributes and functions need to be defined here.